src/VecSim/algorithms/hnsw/hnsw.h

```diff
@@ -1509,27 +1505,24 @@ VecSimQueryResult_List HNSWIndex<DataType, DistType>::rangeQuery(const void *que
         query_data = normalized_blob;
     }
 
-    double originalEpsilon = epsilon_;
+    double epsilon = epsilon_;
     if (queryParams) {
         timeoutCtx = queryParams->timeoutCtx;
         if (queryParams->hnswRuntimeParams.epsilon != 0.0) {
-            epsilon_ = queryParams->hnswRuntimeParams.epsilon;
+            epsilon = queryParams->hnswRuntimeParams.epsilon;
         }
     }
 
     idType bottom_layer_ep = searchBottomLayerEP(query_data, timeoutCtx, &rl.code);
     if (VecSim_OK != rl.code) {
-        epsilon_ = originalEpsilon;
         rl.results = array_new<VecSimQueryResult>(0);
         return rl;
     }
 
     // search bottom layer
-    rl.results = searchRangeBottomLayer_WithTimeout(bottom_layer_ep, query_data, this->epsilon_,
-                                                    radius, timeoutCtx, &rl.code);
+    rl.results = searchRangeBottomLayer_WithTimeout(bottom_layer_ep, query_data, epsilon, radius,
+                                                    timeoutCtx, &rl.code);
 
-    // Restore the default epsilon.
-    epsilon_ = originalEpsilon;
```

Comment:
why did you remove this?

Reply:
because I don't change it in the first place